system/lib/wasmfs/file.h

```diff
@@ -65,30 +64,33 @@ class File : public std::enable_shared_from_this<File> {
 
 protected:
   File(FileKind kind) : kind(kind) {}
+  File(FileKind kind, uint32_t mode) : kind(kind), mode(mode) {}
```

Comment:
Let's add a typedef to the mode, `typedef uint32_t mode_t` perhaps?

edit: grepping suggests that may already exist...

Reply:
Oh yeah that sounds good